### PR TITLE
 Docs: Split up Pull Request Builds into a how-to guide and reference (Diátaxis)

### PR DESCRIPTION
**Pull request by benjaoming**
Refs #9746 

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--9679.org.readthedocs.build/en/9679/

- Developer's documentation (`dev`): https://dev--9679.org.readthedocs.build/en/9679/

### CODE REVIEW

**Review by humitos — DISMISSED** [review]
"Requesting changes" here until we decide the next steps in the conversation we are having in https://github.com/readthedocs/readthedocs.org/pull/9676

Note for the future: we are renaming a file, so that we will require a redirect in place.

**Comment by benjaoming** [comment]
> Note for the future: we are renaming a file, so that we will require a redirect in place.

No files are renamed here, the old file stays in place and a new file is added.

We have in the overall plan for this project to not remove any files, so we don't have to become distracted with file names, folder names nor redirects.

File: docs/user/guides/administrators.rst [hunk 1]
```diff
@@ -19,3 +19,4 @@ have a look at our :doc:`/tutorial/index`.
    deprecating-content
    pdf-non-ascii-languages
    importing-private-repositories
+   Configure Pull Request Builds <pull-requests>
```
Comment (ericholscher) [inline docs/user/guides/administrators.rst:22]: Seems odd to use a custom title here? If we want another title, let's just change the title of the article.

**Review by ericholscher — APPROVED** [review]

**Comment by benjaoming** [comment]
This is ready and reviewed.

Site-note: Picked up a conversation about finding a repeatable pattern for "Only available on Sphinx" messages. But that can be introduced in another changeset.

Merging this one...

**Pull request merged**